### ISSUE TITLE
0dyseus@WallpaperChangerApplet - various issues with settings.

### ISSUE THREAD

**Issue by smurphos** [Odyseus/CinnamonTools#135]
Hi,

I like the applet very much but it has some serious bugs relating to how it manages it's settings.

1) Cannot delete user created profiles from the applets configuration tool. Tried with the profile emptied of images/folders and with the daemon running and not running.
2) Export configuration to file produces an empty dconf file.
3) Reset configuration settings has no effect.
4) On uninstall of the applet via the GUI applet manager it asks for admin password to delete settings and then completes uninstall. However the settings still persist on reinstall. Tried with a reboot after uninstall. Had to install dconf-cli and use _dconf reset -f /org/cinnamon/applets_/ from the terminal to clear the settings.

System:    Kernel: 4.10.0-30-generic x86_64 (64 bit gcc: 5.4.0)
                 Desktop: Cinnamon 3.4.4 (Gtk 3.18.9-1ubuntu3.3) dm: lightdm
                 Distro: Linux Mint 18.2 Sonya

Thanks

**Comment by Odyseus** [comment]
Hello, @smurphos.

1. *Cannot delete user created profiles...*.
    - I can reproduce this. Trying to fix it.
2. *Export configuration to file produces an empty dconf file.*
    - To export the settings, two things needs to be met. The settings schema has to be globally installed and the `dconf` command needs to be available.
        1. The schema should be automatically installed if the applet is installed from Cinnamon's applet manager. If not, it can be manually installed by running the command `./settings.py install-schema` from a terminal opened inside the applet folder. The same can be done to manually uninstall it by running the `./settings.py remove-schema` command.
        2. The `dconf` command is used by the applet to reset its settings (as you discovered and mentioned in point **4**). 
3. *Reset configuration settings has no effect.*
    - The settings schema has to be globally installed. See point **2.a**.
4. *On uninstall of the applet...the settings still persist on reinstall*
    - The settings schema needs to be installed before one is able to uninstall it. See point **2.a**.

Points 2, 3 and 4 could have been avoided if I had not forgotten to add that information on the applet help file, just like I did with the help file for **Cinnamon Tweaks** extension, which uses the same settings system as the **Wallpaper Changer** applet. You can read it [in the on-line help file for that extension](https://odyseus.github.io/CinnamonTools/help_files/0dyseus@CinnamonTweaks.html) (the section called **Extension's settings window**).

I will try to sort out everything for the next version of the applet. Thanks for your feedback. :+1:

**Comment by smurphos** [comment]
Thanks. I've been using the applet manager for download, install & removal and the schema is not being installed or removed although it is bringing up a gksudo window when I remove the applet- but not when I install or add it to the panel.  

After however removal the schema is still there if I check with dconf command.

Calling the functions directly via the terminal to install or remove the schema is working though as is export and reset once it is installed.

Thanks again and glad to help troubleshoot.

**Comment by Odyseus** [comment]
Hello, @smurphos.

Already fixed all issues. You can test them by installing [the applet manually](https://odyseus.github.io/CinnamonTools/pkg/0dyseus@WallpaperChangerApplet.tar.gz). Or you can just wait for me to update the Spices website (tomorrow or the day after at the latest).

It's worth noticing that I also don't get a prompt to install the settings schema when I install or update the applet from the Spices website. This is an anomaly. I will take a deeper look at that when I have the time. Luckily, I always add the `schemas/gschemas.compiled` file to the applet. Without that file, and without the settings schema installed globally, the settings window would never open.

**Comment by Radek71** [comment]
@Odyseus 

I would like to update translation, but seems there is small confuse in wording. Could you please update new added strings - use word "applet" instead of "extension".

thanks a lot

radek

**Comment by Odyseus** [comment]
Hello, @Radek71.

I did that on purpose. In theory, a Cinnamon applet is an *extension* for Cinnamon. I used the word *extension* as the generic meaning of the term. I did this because the **Cinnamon Tweaks** and **Multi Translator** extensions also has the same text describing their settings handling. And to avoid for the translators to deal with almost the same text that needs to be translated differently, I just *reused* the same strings.

But I think that the most sensible thing to do would be to simply use the term *xlet* as a generic term, which I already use in all the help files anyway.
